container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,357 @@
+// Copyright 2016 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+////////////////////////////////////////////////
+// This package manipulates v2.2 image configuration metadata.
+// It writes out both a config file and a manifest for the v2.2 image.
+
+package compat
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+var (
+	emptyFile = emptySHA256Digest([]byte(""))
+)
+
+const (
+	// defaultProcArch is the default architecture type based on legacy create_image_config.py.
+	defaultProcArch = "amd64"
+	// defaultTimeStamp is the unix epoch 0 time representation in 32 bits.
+	defaultTimestamp = "1970-01-01T00:00:00Z"
+)
+
+// emptySHA256Digest returns the sha256 sum of an empty digest.
+func emptySHA256Digest(emptyByte []byte) (empty string) {
```

Comment:
This function shouldn't take an argument. Right now, this function caculates the digest of any given blob of bytes so it would be named `SHA256Digest`. So:
1. Change the function to not accept an argument.
2. Keep the name.
3. Change the impl to always return the digest of an empty byte slice.
4. Call this function instead of `emptyFile`.